lib/keymap-helpers.nix

```diff
@@ -107,9 +107,10 @@ rec {
           (lib.optionalAttrs (isAttrs key || key) {
             key = lib.mkOption (
               {
-                type = types.str;
+                type = with types; nullOr str;
```

Comment:
Hm, I'm not convinced we _always_ want this to be nullable? E.g. in the top-level `keymaps` option, it wouldn't make sense to have a mapping with a null lhs.

Reply:
Well, it needs to be something to access it for conditional mappings like in the test case where we don't want to set it sometimes. If it's a string we have to determine if it's valid to be a keymap, whereas if it's null we know to skip it.

Reply:
Would that make more sense if the keymap as a whole was nullable?

I just struggle to understand having a partial keymap, where we have a action, mode, etc, but no key.